src/app/hooks/useClickTracker/index.test.jsx

```diff
@@ -0,0 +1,322 @@
+/* eslint-disable no-console */
+/* eslint-disable react/prop-types */
+import React from 'react';
+import { renderHook } from '@testing-library/react-hooks';
+import { render, act } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import useClickTracker from '.';
+import pidginData from './fixtureData/tori-51745682.json';
+import { RequestContextProvider } from '#contexts/RequestContext';
+import { ServiceContextProvider } from '#contexts/ServiceContext';
+import { STORY_PAGE } from '#app/routes/utils/pageTypes';
+
+beforeEach(() => {
+  jest.clearAllMocks();
+  console.error = jest.fn();
+});
+
+const urlToObject = url => {
+  const { origin, pathname, searchParams } = new URL(url);
+
+  return {
+    origin,
+    pathname,
+    searchParams: Object.fromEntries(searchParams),
+  };
+};
+
+delete window.location;
+window.location = { href: 'http://bbc.com/pidgin/tori-51745682' };
```

Comment:
```suggestion
window.location = {
  href: 'http://bbc.com/pidgin/tori-51745682',
  assign: jest.fn(),
};
```

Reply:
Actually, never mind, your comment below has fixed it, cheers!

Reply:
No worries! I'm happy to explain on a call if you like. Also, a bit of an oversight in the view hook too, but we should restore the original functionality of window.location when we've done testing - just to be sure there are no side-effects when people add more tests. Something like this would work https://remarkablemark.org/blog/2018/11/17/mock-window-location/#solution